Libraries/LibWeb/Editing/Internal/Algorithms.cpp

```diff
@@ -24,6 +24,98 @@
 
 namespace Web::Editing {
 
+// https://w3c.github.io/editing/docs/execCommand/#block-extend
+GC::Ref<DOM::Range> block_extend_a_range(DOM::Range& range)
+{
+    // 1. Let start node, start offset, end node, and end offset be the start and end nodes and offsets of range.
+    GC::Ptr<DOM::Node> start_node = range.start_container();
+    auto start_offset = range.start_offset();
+    GC::Ptr<DOM::Node> end_node = range.end_container();
+    auto end_offset = range.end_offset();
+
+    // 2. If some inclusive ancestor of start node is an li, set start offset to the index of the last such li in tree
+    //    order, and set start node to that li's parent.
+    auto ancestor = start_node;
+    do {
+        if (is<DOM::Element>(*ancestor) && static_cast<DOM::Element&>(*ancestor).local_name() == HTML::TagNames::li) {
+            start_offset = ancestor->index();
+            start_node = ancestor->parent();
+            break;
+        }
+        ancestor = ancestor->parent();
+    } while (ancestor);
+
+    // 3. If (start node, start offset) is not a block start point, repeat the following steps:
+    if (!is_block_start_point(*start_node, start_offset)) {
+        do {
+            // 1. If start offset is zero, set it to start node's index, then set start node to its parent.
+            if (start_offset == 0) {
+                start_offset = start_node->index();
+                start_node = start_node->parent();
+            }
+
+            // 2. Otherwise, subtract one from start offset.
+            else {
+                --start_offset;
+            }
+
+            // 3. If (start node, start offset) is a block boundary point, break from this loop.
+        } while (!is_block_boundary_point(*start_node, start_offset));
+    }
+
+    // 4. While start offset is zero and start node's parent is not null, set start offset to start node's index, then
+    //    set start node to its parent.
+    while (start_offset == 0 && start_node->parent()) {
+        start_offset = start_node->index();
+        start_node = start_node->parent();
+    }
+
+    // 5. If some inclusive ancestor of end node is an li, set end offset to one plus the index of the last such li in
+    //    tree order, and set end node to that li's parent.
+    ancestor = end_node;
+    do {
+        if (is<DOM::Element>(*ancestor) && static_cast<DOM::Element&>(*ancestor).local_name() == HTML::TagNames::li) {
```

Comment:
You should be able to do `is<HTMLLIElement>()` here.

Reply:
Good find! I've found a couple of other places too.